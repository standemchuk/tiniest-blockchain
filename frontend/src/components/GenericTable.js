import React, { PureComponent } from 'react'
import Grid from '@material-ui/core/Grid'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Typography from '@material-ui/core/Typography'

export default class GenericTable extends PureComponent {
  render () {
    const { data, headers } = this.props
    if (!data || !data.length) {
      return (
        <Grid container>
          <Grid item xs={12}>
            <Typography style={{ marginLeft: 30 }} variant='subheading'>No data</Typography>
          </Grid>
        </Grid>
      )
    }
    return (
      <Table>
        <TableHead>
          <TableRow>
            {headers && headers.map(header => (
              <TableCell key={header}>{header}</TableCell>
            ))}
          </TableRow>
        </TableHead>
        <TableBody>
          {data.map((item, index) => (
            <TableRow key={index.toString()}>
              {Object.keys(item).map(key => (
                <TableCell key={key}>{item[key]}</TableCell>
              ))}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    )
  }
}
