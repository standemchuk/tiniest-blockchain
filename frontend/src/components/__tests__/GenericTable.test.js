import React from 'react'
import renderer from 'react-test-renderer'
import Typography from '@material-ui/core/Typography'

import GenericTable from '../GenericTable'

it('renders correctly', () => {
  const props = {
    data: [ {test1: 1, test2: 2} ],
    headers: ['Test 1', 'Test 2']
  }
  const tree = renderer
    .create(
      <GenericTable {...props} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a text warning when no data is supplied', () => {
  const tree = renderer
    .create(<GenericTable />)

  const text = tree.root.findByType(Typography)

  expect(text.props.children).toBe('No data')
})
