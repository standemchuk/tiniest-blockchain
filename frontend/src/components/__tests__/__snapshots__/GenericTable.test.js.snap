// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="MuiTable-root-1"
>
  <thead
    className="MuiTableHead-root-2"
  >
    <tr
      className="MuiTableRow-root-3 MuiTableRow-head-6"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Test 1
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Test 2
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-16"
  >
    <tr
      className="MuiTableRow-root-3"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
        scope={undefined}
      >
        1
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
        scope={undefined}
      >
        2
      </td>
    </tr>
  </tbody>
</table>
`;
